### ISSUE TITLE
Lock detection 

### ISSUE THREAD

**Issue by vakandi** [vakandi/zoomscript#1]
Checking process like User event Manager, loginw of loginwindow doesn't work, need to try another way to detect lock screen on MacOs

**Comment by vakandi** [comment]
Fixed it after testing every process linked to the lock or sleeping screen (both do the same thing we want) without success, i used this function:
`function screenIsUnlocked { [ "$(/usr/libexec/PlistBuddy -c "print :IOConsoleUsers:0:CGSSessionScreenIsLocked" /dev/stdin 2>/dev/null <<< "$(ioreg -n Root -d1 -a)")" != "true" ] && return 0 || return 1; }`
Which returning the right value by **access the system preferences settings without root or sudo access**

**Comment by vakandi** [comment]
The function uses the command ioreg to get a property list (plist) from the I/O Registry, which is a database of device drivers and other information about the hardware and software of a computer.
The -n option specifies the name of the registry entry to query and the -d1 option specifies the depth of the search. The -a option to display all properties of the entry.

The output of the ioreg command is piped to /usr/libexec/PlistBuddy command which is a tool for manipulating plist files. The -c option specifies a command to execute on the plist file which is passed to PlistBuddy as a standard input. The command print :IOConsoleUsers:0:CGSSessionScreenIsLocked is used to print the value of the key CGSSessionScreenIsLocked, which indicates if the screen is locked or not.

The output of the PlistBuddy command is saved to a variable and it's compared with "true" string, if the output is equal to "true" then the screen is locked, and the function returns 1(true) otherwise it returns 0(false).

In this script, the && operator is used to run the command after it, only if the command before it was successful. The || operator is used to run the command after it, only if the command before it failed.

In summary, this function uses the ioreg and PlistBuddy command to check the screen lock status of the computer and returns 0 if the screen is unlocked and 1 if the screen is locked.
